lib/helpers/wp-cli-db/class-wp-cli-db.php

```diff
@@ -36,6 +45,26 @@ public function early_init() {
 		WP_CLI::add_command( 'db', __NAMESPACE__ . '\VIP_DB_Command' );
 	}
 
+	/**
+	 * Since there is an incompat with HyperDB, we need to force the --all-tables flag for `wp db tables`
+	 */
+	public function before_invoke_db_tables() {
+		if ( isset( $this->assoc_args['all-tables'] ) && ( 'true' === $this->assoc_args['all-tables'] || true === $this->assoc_args['all-tables'] ) ) {
```

Comment:
Here we can define the property on `$GLOBALS['wpdb']`

Reply:
Alternate approach: https://github.com/Automattic/vip-go-mu-plugins/pull/4376